lib/quantum/date_library/calendar.ex

```diff
@@ -24,11 +24,12 @@ if Code.ensure_compiled?(Calendar.DateTime) do
 
     alias Calendar.DateTime, as: CalendarDateTime
 
-    @spec utc_to_tz(NaiveDateTime.t, String.t) :: NaiveDateTime.t | no_return
-    def utc_to_tz(date, tz) do
-      date
-      |> DateTime.from_naive!("Etc/UTC")
-      |> CalendarDateTime.shift_zone!(tz)
+    @spec tz_to_utc(NaiveDateTime.t, String.t) :: NaiveDateTime.t | no_return
+    def tz_to_utc(date, tz) do
+      {:ok, tz_date} = CalendarDateTime.from_naive(date, tz)
+
+      tz_date
+      |> CalendarDateTime.shift_zone!("Etc/UTC")
```

Comment:
This looks correct, but I'm still a little worried. Might need some extensive test cases to prove that it works as expected. Or maybe I'm too stupid to see that it's "obviously" correct :) (I know you just haven't got to the tests yet, not yelling at you telling you to write tests, just saying this could use a few good tests) 😸